ls1intum/Artemis
# Bugfix/bamboo special char replacement

### Checklist

### Motivation and Context


This should fix the problems described in [ARTEMIS-406](https://jirabruegge.in.tum.de/browse/ARTEMIS-406?filter=-1). If a new programming exercise had special character in its name, we couldn't create a build plan in Bamboo, because we were using the full exercise name in the Bamboo project name.

### Description

As described in the ticket, I am removing every "illegal" character from the Bamboo project name before creating the build plans.

### Steps for Testing


1. Log in to ArTEMiS
2. Navigate to Course Administration
3. Click on "Exercise" for one of the courses and create a new programming exercise.
4. Use any exercise name and include any of the following characters: "[, ], {, }, <, >, :, @, /, &, %, \, !, |, #, $, *, ; or ~"
5. ~Tick "Publish Build Plan" at the bottom of the page to create the Bamboo plans.~
6. The exercise should be created successfully and the project name in Bamboo shouldn't contain any of the special characters.

### Screenshots


![bamboo_special_chars](https://user-images.githubusercontent.com/27979674/61596300-c6b1da00-ac01-11e9-83a9-2921ec62c989.gif)
